src/estimation/include/iDynTree/Estimation/BerdySparseMAPSolver.h

```diff
@@ -68,14 +72,14 @@ namespace iDynTree {
                                                    const JointDOFsDoubleArray& jointsVelocity,
                                                    const FrameIndex floatingFrame,
                                                    const Vector3& bodyAngularVelocityOfSpecifiedFrame,
-                                                   const VectorDynSize& measurements);
-
-        bool doEstimate();
+                                                   const VectorDynSize& measurements,
+                                                   const Transform& w_H_b = iDynTree::Transform::Identity(),
+                                                   const HierarchialBerdyTask& task = HierarchialBerdyTask::FULL_DYNAMICS);
 
-        void getLastEstimate(iDynTree::VectorDynSize& lastEstimate) const;
+        bool doEstimate(const HierarchialBerdyTask& task = HierarchialBerdyTask::FULL_DYNAMICS);
```

Comment:
```suggestion
        bool doEstimate(const HierarchialBerdyTask task = HierarchialBerdyTask::FULL_DYNAMICS);
```
No need to have references (i.e. pointers) when you just need to pass an enum value (see https://stackoverflow.com/questions/26552481/which-is-faster-pass-by-reference-vs-pass-by-value-c ).